dataRetriever/dataPool/headersCache/headersNonceCache.go

```diff
@@ -0,0 +1,252 @@
+package headersCache
+
+import (
+	"bytes"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data"
+)
+
+type headerListDetails struct {
+	headerList []headerDetails
+	timestamp  time.Time
+}
+
+type headerInfo struct {
+	headerNonce   uint64
+	headerShardId uint32
+}
+
+type headersNonceCache struct {
+	hdrNonceCache map[uint32]*headersMap
+
+	headersByHash *headersHashMap
+	hdrsCounter   *headersCounter
+
+	mutHeadersMap      sync.RWMutex
+	numHeadersToRemove int
+
+	maxHeadersPerShard int
+	canDoEviction      map[uint32]chan struct{}
```

Comment:
Do not use channels for this purpose, please. Please see the comment below for `tryToDoEviction`.

Reply:
channels have been removed